### DISCUSSION TITLE
Some improvements of submarine death at extreme depth

### DISCUSSION THREAD

**Discussion by MaaiG** [FakeFishGames/Barotrauma#7720] [Ideas and suggestions]
The submarine must don't die instantly in deep water. There must be events that foreshadow and you must be able to prevent catastrophe.

There must be a range to stop or leave the boat. The depth sensor is not enough. Audio and video effects are needed to determine the depth limit.
- The sound of the maximum load on the shell
- Leaking
- Crackling bulkhead,  burst etc
- The bulkheads of the submarine must not be destroyed all at once. Just water pressure bulkheads in the flooded compartments.

https://youtu.be/MimdpQJmCfM?t=213

**Submarine editor**
It’s not the boat that’s dying. It’s the crew that’s dying. If the boat passes the point of no return by inertia and then goes back to a tolerable depth. The crew will be dead but the submarine is fine.

A simple automatic circuit stops the dive and returns the submarine to an acceptable depth.
Simulate movement inertia as a result of the completion of the flood control:
https://user-images.githubusercontent.com/93348499/145681153-bbf4e18d-c5a5-4157-8ab8-0b2e1184971f.mp4

**Comment by cshingle** [comment]
I really like the idea of adding additional sound effects. I think periodic popping or cracking sounds happening when the hull takes damage from being below crush depth would be a great addition.

**Reply by MaaiG** [reply]
> I think periodic popping or cracking sounds happening when the hull takes damage from being below crush depth would be a great addition.

However It may be interesting to add sounds made by creatures colliding and collapsing ice caps, unknown sounds requiring examination or greater caution. Some sounds can only be heard by sonar.

When I was playing "SOMA", I was saddened by the lack of sound of living things. Walking on the bottom of the ocean and looking up, I really wanted to hear the sounds of whales and maybe even see a huge shadow from the creature. This would provide an even greater atmosphere and immersion in the game.

The game should be interesting even if your eyes closed. :)

**Reply by MateusJK15** [reply]
This link leads to a video i uploaded to google drive (github didn't let me upload the video here because 90MB is "too big")
https://drive.google.com/file/d/1Qjny0IfOaNr5U9FFIHgj5Zn4j0inSN7b/view?usp=sharing
Pay attention to when i reach 1100ft and 1360ft, the sound comes from the perspective of the crew inside the sub (the game in this video is Dangerous Waters)

**Reply by leonard84** [reply]
I think this scene does describe it pretty well https://www.youtube.com/watch?v=cUMw4mPuXcc

**Comment by MaaiG** [comment]
In addition, the depth indicator of the navigation terminal does not give an idea of how deep the dive is dangerous for your submarine.
A simple numerical pointer is not sufficiently informative.
The depth indicator shall display a dangerous depth depending on the current shell improvements.
![Image 001_20-02a](https://user-images.githubusercontent.com/93348499/153046324-46bf27c7-27d1-4935-8f15-1eed3a1e1092.png)
The death of the submarine at critical depths may not occur instantaneously at dangerous depths, but may have a range within which the death of the submarine will depend on many internal and external factors. For example, hull condition, immersion rate, percentage of flooded compartments, undercurrents, external impact etc. This will add a flavor of chance, luck or perhaps skill if you feel your ship and will be listen well to it.
![Image 001_20-151b](https://user-images.githubusercontent.com/93348499/153047756-383813fb-9f7a-4ef2-9df4-77d4b33cf46d.png)

**Reply by GoodGuyJimmy** [reply]
Yeah, if they replaced the current numerical counter with that clock-like depth meter from the first image, it would not only look a lot cooler, but would also give you some necessary information. I wish Barotrauma's UI was more like Highfleet's UI in certain areas such as the submarine control terminal.

**Comment by Prolbo** [comment]
Yes, more foreshadowing would be nice. Audio part definitely can use some improvements.

**Comment by Xsayen** [comment]
Hey guys, I was about to make a new discussion. I'm so glad I double checked.
I just wanted to add my own input and talk about the technical side of things. I would really like if we changed the current boring crush depth mechanics.

Basically when a submarine goes under the depth limit, instead of taking damage as depth increases an invisible buffer number would go down. As it ticks down to zero the menacing sounds of buckling and groaning metal will increase in intensity. I can imagine how anxiety of the crew grows as the pounding gets louder. And when the value reaches zero the whole hull implodes immediately flooding the whole ship. This is correct to the real world plus it also makes sense as the outer shell is the most resistant anyway, so if it gives way nothing would stop water from crushing the weak inner walls.

The buffer or especially the time before you implode could be influenced by hull upgrades but I'm not sure about the balance. Same with the function of the decrease rate. It could be linear with depth or exponential or any other equation. There could also be some randomness to when the implosion happens so it's not always the same moment every time.
